docs/contribute.rst

```diff
@@ -423,10 +423,54 @@ There is a nice way to run tests by disallowing network calls. This can help us
 You can add ``--disable-socket`` to any pytest call and it will fail immediately for any network calls. You will probably need to also add ``--allow-hosts=127.0.0.1`` if the tests makes local network calls to the rotki api. This way you can discover all network calls and mock them.
 
 Mocking should happen with one of the following ways
-1. Using common fixtures for data mocking as started and shown `here <https://github.com/rotki/rotki/pull/5269`__. Read the PR description to get an idea.
+1. Using common fixtures for data mocking as started and shown `here <https://github.com/rotki/rotki/pull/5269`__ . Read the PR description to get an idea.
 2. Using test specific mocking.
 3. For repeatable calls that would always return the same response from the network use the vcr.py approach.
 
+After 1.26.3 we have introduced VCR to mock network queries in some tests trying to improve the speed of our slower ones. VCR works by generating a `yaml` file that has information about the request made and then for every request that happens in the test VCR tries to match it to one of the recorded. We have already some pre-recorded cassettes (the name used by VCR for those yaml files) and they are available at `github <https://github.com/rotki/test-caching>`__.
+In a fresh run this repo will be cloned and then the cassettes will be replayed, this happens in the path set by the ``vcr_cassette_dir`` fixture that also sets the directory where the cassettes are located.
+
+Locally cassettes are only read and never written to prevent unexpected behaviour during testing. To record a new test we provide a make rule that allows it ``create-cassette``. First we need to mark the test as a VCR test with the pytest directive
```

Comment:
```suggestion
Locally cassettes are only read and never written to prevent unexpected behaviour during testing. To record a new test we provide a make rule that allows it called ``create-cassette``.

First we need to mark the test as a VCR test with the pytest directive
```